code/datums/components/item_equipped_movement_rustle.dm

```diff
@@ -46,22 +44,21 @@
 	if(!(slot & our_item.slot_flags))
 		return
 	SIGNAL_HANDLER
-	holder = equipper
-	RegisterSignal(holder, COMSIG_MOVABLE_MOVED, PROC_REF(try_step), override = TRUE)
+	RegisterSignal(equipper, COMSIG_MOVABLE_MOVED, PROC_REF(try_step), override = TRUE)
 
 /datum/component/item_equipped_movement_rustle/proc/on_unequip(datum/source, mob/equipper, slot)
```

Comment:
unrelated to ur fix but these args dont match the ones sent by the `COMSIG_ITEM_DROPPED`  signal . U could register a `COMSIG_MOB_UNEQUIPPED_ITEM` signal on the mob inside `on_equip()`  and have it call this instead, or if ud rather keep the dropped signal u could just fix the args.